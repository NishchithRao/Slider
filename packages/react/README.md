# Slider React

A React wrapper for the slider library.

## Installation

```bash
npm install @slider/react
```

## Usage

Wrap your elements with the Slider container, provide a transition and you're good to go!.

```javascript
import { SliderContainer } from "@slider/react";
() => {
  return (
    <>
      <SliderContainer transition="default">
        <div>Slide1</div>
        <div>Slide2</div>
        <div>Slide3</div>
      </SliderContainer>
    </>
  );
};
```
