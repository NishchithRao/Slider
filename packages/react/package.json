{
  "name": "@slider/react",
  "version": "0.0.0",
  "description": "> TODO: description",
  "author": "Nishchith.Rao <nischitrao5@gmail.com>",
  "homepage": "",
  "license": "ISC",
  "main": "dist/index.cjs",
  "module": "dist/index.mjs",
  "typings": "dist/index.d.ts",
  "peerDependencies": {
    "react": ">=16",
    "react-dom": ">=16"
  },
  "dependencies": {
    "@slider/core": "*"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c",
    "postbuild": "sass --no-source-map src/index.scss dist/slider-react.css"
  }
}
