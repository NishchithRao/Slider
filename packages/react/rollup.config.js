const commonConfig = require("../../rollup.config");
const pkg = require("./package.json");

module.exports = {
  ...commonConfig,
  input: "./src/index.ts",
  output: [
    {
      file: pkg.module,
      format: "esm",
      exports: "named",
    },
    {
      file: pkg.main,
      format: "cjs",
      exports: "named",
    },
  ],
  external: ["react", "react-dom"],
};
