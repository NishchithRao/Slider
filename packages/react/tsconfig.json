{
    "extends": "../../tsconfig",
    "compilerOptions": {
        "jsx": "react-jsx",
        "outDir": "dist",
        "module": "ESNext",
        "target": "ES5"
    },
    "include": ["src"],
    "exclude": ["node_modules"]
}