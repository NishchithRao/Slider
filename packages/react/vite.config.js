import react from '@vitejs/plugin-react';
import { sharedConfig } from '../../vite.config';
import pkgJSON from './package.json';

/** @type {import('vite').UserConfig} */
export default {
  plugins: [react(), ...sharedConfig(pkgJSON).plugins],
  ...sharedConfig(pkgJSON)
};
