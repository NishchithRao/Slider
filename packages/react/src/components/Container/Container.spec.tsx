import { render } from "@testing-library/react";
import { SliderContainer } from "..";

describe("Slider Container", () => {
  it("should render", () => {
    const result = render(
      <SliderContainer>
        <div>Slide 1</div>
        <div>Slide 2</div>
        <div>Slide 3</div>
      </SliderContainer>
    );

    expect(result.container.innerHTML).toMatchSnapshot();
  });
});
