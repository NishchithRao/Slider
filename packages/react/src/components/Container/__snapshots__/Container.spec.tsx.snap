// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Container should render 1`] = `"<div class="Container"><div data-index="0"><div>Slide 1</div></div><div data-index="1"><div>Slide 2</div></div><div data-index="2"><div>Slide 3</div></div></div>"`;
